hcl2template/types.packer_config.go

```diff
@@ -104,12 +106,26 @@ func (p *Parser) getBuilds(cfg *PackerConfig) ([]packer.Build, hcl.Diagnostics)
 			if moreDiags.HasErrors() {
 				continue
 			}
-			provisioners, moreDiags := p.getCoreBuildProvisioners(build.ProvisionerBlocks, cfg.EvalContext(), generatedVars)
+
+			// If the builder has provided a list of to-be-generated variables that
+			// should be made accessible to provisioners, pass that list into
+			// the provisioner prepare() so that the provisioner can appropriately
+			// validate user input against what will become available. Otherwise,
+			// only pass the default variables, using the basic placeholder data.
+			generatedPlaceholderMap := packer.BasicPlaceholderData()
+			if generatedVars != nil {
+				for _, k := range generatedVars {
+					generatedPlaceholderMap[k] = fmt.Sprintf("Build_%s. "+
+						common.PlaceholderMsg, k)
+				}
+			}
```

Comment:
Moving this up so it can be used by post-processors as well. Also rename `Generated_`  to  `Build_` to keep consistent with CoreBuild.Prepare(...) https://github.com/hashicorp/packer/blob/master/packer/build.go#L168